File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Go/s893190600_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Solver struct {
	in  *MyInput
	out *bufio.Writer
}

func (s *Solver) solve() {
	H := s.in.nextInt()
	W := s.in.nextInt()
	C := make([][]rune, H+2)
	for i := range C {
		C[i] = make([]rune, W+2)
	}
	for i := 0; i < H; i++ {
		line := s.in.nextString()
		for j, char := range line {
			C[i+1][j+1] = char
		}
	}
	if H == 1 && W == 1 {
		s.prn("No")
		return
	}
	for i := 1; i <= H; i++ {
		for j := 1; j <= W; j++ {
			if C[i][j] == '#' && C[i-1][j] != '#' && C[i+1][j] != '#' &&
				C[i][j-1] != '#' && C[i][j+1] != '#' {
				s.prn("No")
				return
			}
		}
	}
	s.prn("Yes")
}

func (s *Solver) prn(str string) {
	s.out.WriteString(str + "\n")
}

func (s *Solver) prr(str string) {
	s.out.WriteString(str)
}

type MyInput struct {
	scanner *bufio.Scanner
}

func NewMyInput(reader *bufio.Reader) *MyInput {
	scanner := bufio.NewScanner(reader)
	scanner.Split(bufio.ScanBytes) // Read byte by byte
	return &MyInput{scanner}
}

func (m *MyInput) nextString() string {
	var line strings.Builder
	for m.scanner.Scan() {
		char := m.scanner.Text()
		if char == "\n" {
			break
		}
		line.WriteString(char)
	}
	return line.String()
}

func (m *MyInput) nextInt() int {
	str := m.nextString()
	num, _ := strconv.Atoi(str)
	return num
}

func (m *MyInput) nextLong() int64 {
	str := m.nextString()
	num, _ := strconv.ParseInt(str, 10, 64)
	return num
}

func (m *MyInput) nextDouble() float64 {
	str := m.nextString()
	num, _ := strconv.ParseFloat(str, 64)
	return num
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	in := NewMyInput(reader)
	solver := &Solver{in: in, out: writer}
	solver.solve()
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Python/s893190600_comments_removed.py
import sys
from collections import defaultdict, deque

class MyInput:
    def __init__(self, input_stream=sys.stdin):
        self.input_stream = input_stream
        self.buffer = []
        self.pos = 0

    def next_line(self):
        while not self.buffer:
            self.buffer = self.input_stream.readline().strip().split()
        return self.buffer.pop(0)

    def next_int(self):
        return int(self.next_line())

    def next_long(self):
        return int(self.next_line())

    def next_double(self):
        return float(self.next_line())

    def next_string(self):
        return self.next_line()

class Solver:
    def __init__(self, input_stream, output_stream):
        self.inp = MyInput(input_stream)
        self.out = output_stream

    def solve(self):
        H = self.inp.next_int()
        W = self.inp.next_int()
        C = [['' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(H):
            line = self.inp.next_string()
            for j in range(W):
                C[i + 1][j + 1] = line[j]

        if H == 1 and W == 1:
            self.out.write("No\n")
            return

        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i-1][j] != '#' and C[i+1][j] != '#' and C[i][j-1] != '#' and C[i][j+1] != '#':
                    self.out.write("No\n")
                    return

        self.out.write("Yes\n")

if __name__ == "__main__":
    solver = Solver(sys.stdin, sys.stdout)
    solver.solve()

# End of Code
